debugtools/DDR_VM/src/com/ibm/j9ddr/tools/ddrinteractive/CacheMemorySource.java

```diff
@@ -0,0 +1,166 @@
+ * is available at https://www.apache.org/licenses/LICENSE-2.0.                                                                                                              
+ *                                                                                                                                                                            * This Source Code may also be made available under the following                                                                                                           
+ * Secondary Licenses when the conditions for such availability set                                                                                                          
+ * forth in the Eclipse Public License, v. 2.0 are satisfied: GNU                                                                                                            
+ * General Public License, version 2 with the GNU Classpath                                                                                                                  
+ * Exception [1] and GNU General Public License, version 2 with the                                                                                                          
+ * OpenJDK Assembly Exception [2].                                                                                                                                           
+ *                                                                                                                                                                           
+ * [1] https://www.gnu.org/software/classpath/license.html                                                                                                                   
+ * [2] http://openjdk.java.net/legal/assembly-exception.html                                                                                                                 
+ *                                                                                                                                                                           
+ * SPDX-License-Identifier: EPL-2.0 OR Apache-2.0 OR GPL-2.0 WITH Classpath-exception-2.0 OR LicenseRef-GPL-2.0 WITH Assembly-exception                                       
+ *******************************************************************************/
+package com.ibm.j9ddr.tools.ddrinteractive;
+
+import java.nio.ByteOrder;
+import java.nio.ByteBuffer;
+import java.nio.Buffer;
+import java.lang.reflect.Field;
+import com.ibm.j9ddr.corereaders.memory.MemoryFault;
+import com.ibm.j9ddr.corereaders.Platform;
+import com.ibm.j9ddr.corereaders.memory.AbstractMemory;
+import com.ibm.j9ddr.corereaders.memory.IMemorySource;
+import com.ibm.j9ddr.corereaders.memory.IMemoryRange;
+import com.ibm.j9ddr.corereaders.memory.ProtectedMemoryRange;
+import java.io.IOException;
+import java.nio.BufferUnderflowException;
+import sun.nio.ch.DirectBuffer;
```

Comment:
This should not access `sun.nio.ch.DirectBuffer` - in Java11+, it's not exported, not even from `jdk.unsupported`.